strawberryfields/tdm/tdmprogram.py

```diff
@@ -83,63 +84,136 @@ def validate_measurements(circuit, N):
     return spatial_modes
 
 
-def input_check(args, copies):
+def input_check(args):
     """Checks the input arguments have consistent dimensions.
 
     Args:
         args (Sequence[Sequence]): sequence of sequences specifying the value of the parameters
-        copies (int): number of times the circuit should be run
-
     """
     # check if all lists are of equal length
     param_lengths = [len(param) for param in args]
     if len(set(param_lengths)) != 1:
         raise ValueError("Gate-parameter lists must be of equal length.")
 
-    # check copies
-    if not isinstance(copies, int) or copies < 1:
-        raise TypeError("Number of copies must be a positive integer.")
-
 
-def _get_mode_order(num_of_values, N):
+def _get_mode_order(num_of_values, N, timebins):
```

Comment:
Would we eventually go with the `timebins` naming in SF (opposed to `temporal_modes`)?

Reply:
Don't think we decided upon anything. I feel like, at least, timebins is what is best used inside of `tdmprogram.py` since it's been using that all along, while temporal modes have been used for Blackbird and thus `sf.io` and the device spec also. I think it's fine keeping it like this for, but I guess this is still up for discussion since it might be a bit confusing for users.

Reply:
Think this will be confusing in the future in the code base.